controllers/glance_controller.go

```diff
@@ -832,6 +824,45 @@ func (r *GlanceReconciler) ensureRegisteredLimits(
 	return nil
 }
 
+// ensurePVC - Creates the PVCs required by Glance to start the GlanceAPI Pods
+func (r *GlanceReconciler) ensurePVC(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.Glance,
+	serviceLabels map[string]string,
+) (ctrl.Result, error) {
+	// Define a new PVC object
+	localPvc := pvc.NewPvc(
+		glance.Pvc(instance, serviceLabels, glance.PvcLocal),
+		time.Duration(5)*time.Second,
+	)
+
+	ctrlResult, err := localPvc.CreateOrPatch(ctx, h)
+
+	if err != nil {
+		return ctrlResult, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Handle an additional PVC creation an ImageCacheSize is provided
+	if len(instance.Spec.ImageCacheSize) > 0 {
+		cachePvc := pvc.NewPvc(
+			glance.Pvc(instance, serviceLabels, glance.PvcCache),
+			time.Duration(5)*time.Second,
+		)
+		ctrlResult, err := cachePvc.CreateOrPatch(ctx, h)
+
+		if err != nil {
+			return ctrlResult, err
+		} else if (ctrlResult != ctrl.Result{}) {
+			return ctrlResult, nil
+		}
+	}
+	// End PVC creation/patch
+	return ctrlResult, nil
```

Comment:
This could be simplified to reuse the `ctrlResult` and `err` variables from L840, and you could also remove the if-statement and just `return ctrlResult, err` always.

Reply:
I agree the if statement is actually redundant as the same block is repeated/evaluated after the execution of this function.
I still need the first block because if the local pvc creation fails, I'd like to return and block the execution, but +1 on removing the second one. And yeah, I can actually reuse the existing variables.
I'll push a quick update for this part.

Reply:
Done

Reply:
Yeah, sorry, I didn't meant to imply removing both if-blocks.  It looks good now!